### PR TITLE
Update the disease service to use BasicSearchController and BaseServi…

### PR DESCRIPTION
**Pull request by sahmad-ebi**
…ce Classes

### CODE REVIEW

File: core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import uk.ac.ebi.uniprot.cv.disease.Disease;
+import uk.ac.ebi.uniprot.cv.keyword.Keyword;
+import uk.ac.ebi.uniprot.parser.tsv.uniprot.NamedValueMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DiseaseEntryMap implements NamedValueMap {
```
Comment (LeonardoGonzales) [inline core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java]: Can you please, Create a UnitTests for this class

File: core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        if (Utils.notEmpty(crossReferences)) {
+            return crossReferences.stream().map(CrossReference::getId).collect(Collectors.joining(", "));
+        } else {
+            return EMPTY_STRING;
+        }
+    }
+
+
+    private String getAlternativeNames(List<String> alternativeNames) {
+        if (alternativeNames != null) {
```
Comment (LeonardoGonzales) [inline core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java]: Use Utils.notEmpty

File: core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            return this.diseaseEntry.getAlternativeNames().stream().collect(Collectors.joining(", "));
```
Comment (LeonardoGonzales) [inline core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java]: I think, if you use String.join(", ", alternativeNames) is simpler. Another detail, you are passing a parameter and not using it to iterate over the alternative names.

Comment (sahmad-ebi) [reply core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java]: thanks.

File: core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java (hunk 1 above)
Comment (sahmad-ebi) [reply core-parser/src/main/java/uk/ac/ebi/uniprot/parser/tsv/disease/DiseaseEntryMap.java]: added. thanks

**Pull request merged**